### PR TITLE
logging for MPI runs

### LINKED ISSUE #166: "fix" logging output on MPI-parallelized runs
When running in parallel with MPI, the logging output should somehow be differently handled than in a serial environment. There come two options to my mind:

1. make all calls to easylogging++ a no-op (via preprocessor guards)
2. use MPI-IO and configure easylogging++ to only write to those MPI-IO files.

Option one is a low-hanging fruit and is easy to realize. The second option is probably the one to go for in the long run but requires some more work to get accomplished.

Comments?

### PR DESCRIPTION
**Pull request by torbjoernk**
This PR tries to fix #166 by writing the logging output to per-rank files and disabling stdout logging.

This also required to disable building of non-MPI-dependent targets with MPI enabled.

### CODE REVIEW

**Comment by memmett** [comment]
Are you sure you want to disable stdout logging as well?  Usually I like to have stdout logging as well as per-rank file based logging.

How about we have both stdout and per-rank logging on by default, and make it really easy to turn them off/on.

File: examples/advection_diffusion/mpi_pfasst.cpp [hunk 1]
```diff
@@ -89,7 +89,10 @@ namespace pfasst
 #ifndef PFASST_UNIT_TESTING
 int main(int argc, char** argv)
 {
-  MPI_Init(&argc, &argv);
```
Comment (andydawson) [inline examples/advection_diffusion/mpi_pfasst.cpp]: The user should be responsible for doing MPI_Init.

File: include/pfasst.hpp [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
       opts();
     }
     config::init();
+    if (with_mpi) {
+#ifdef WITH_MPI
+      MPI_Init(&argc, &argv);
+#else
+      cerr << "PFASST::init() : 'with_mpi' flag used without enabling MPI" << endl;
+#endif
+    }
```
Comment (andydawson) [inline include/pfasst.hpp]: I guess the block above can go if we require the user to call `MPI_Init`.

File: include/pfasst.hpp [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
+                          std::function<void()> logs = nullptr,
+                          bool with_mpi = false)
   {
     if (opts) {
       opts();
     }
     config::init();
+    if (with_mpi) {
+#ifdef WITH_MPI
+      MPI_Init(&argc, &argv);
```
Comment (memmett) [inline include/pfasst.hpp]: The user should be responsible for calling `MPI_Init`.

File: examples/advection_diffusion/mpi_pfasst.cpp [hunk 4]
```diff
@@ -89,7 +89,10 @@ namespace pfasst
...
+  pfasst::init(argc, argv,
```
Comment (memmett) [inline examples/advection_diffusion/mpi_pfasst.cpp]: The user should be responsible for calling `MPI_Init`.

File: include/pfasst.hpp (hunk 2 above)
Comment (memmett) [inline include/pfasst.hpp]: The block above can go (user should call `MPI_Init`).

**Comment by torbjoernk** [comment]
once this is merged I'll open the PR for the JUQUEEN specific changes already residing in my fork

File: include/pfasst.hpp [hunk 5]
```diff
@@ -1,6 +1,9 @@
 #ifndef _PFASST_HPP_
 #define _PFASST_HPP_
 
+#include <iostream>
+using namespace std;
+
```
Comment (memmett) [inline include/pfasst.hpp]: Doesn't look like we need these?

Comment (torbjoernk) [inline include/pfasst.hpp]: my fault. forgot to delete those.

**Pull request merged**